src/Symfony/Component/Serializer/Mapping/AttributeMetadata.php

```diff
@@ -111,15 +109,51 @@ public function getMaxDepth()
      */
     public function setSerializedName(string $serializedName = null)
     {
-        $this->serializedName = $serializedName;
+        $this->addSerializedName($serializedName);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addSerializedName(string $serializedName, array $groups = [])
+    {
+        $this->serializedNames[$serializedName] = $groups;
     }
 
     /**
      * {@inheritdoc}
      */
     public function getSerializedName(): ?string
     {
-        return $this->serializedName;
+        return $this->getSerializedNameForGroups();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSerializedNames(): array
+    {
+        return $this->serializedNames;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSerializedNameForGroups(array $groups = []): ?string
+    {
+        $defaultSerializedName = null;
+
+        foreach ($this->serializedNames as $serializedName => $groupsForSerializedName) {
+            if (!$groupsForSerializedName) {
+                $defaultSerializedName = $serializedName;
```

Comment:
i dont think you can make this assumption :/ any name without groups could be considered default